pre_commit/xargs.py

```diff
@@ -168,9 +206,41 @@ def xargs(
     def run_cmd_partition(
             run_cmd: tuple[str, ...],
     ) -> tuple[int, bytes, bytes | None]:
-        return cmd_fn(
-            *run_cmd, check=False, stderr=subprocess.STDOUT, **kwargs,
-        )
+        if not stream_output:
+            return cmd_fn(
+                *run_cmd, check=False, stderr=subprocess.STDOUT, **kwargs,
+            )
+
+        output = b''
+
+        with stdout_lock:
+            sys.stdout.buffer.write(b'\n')
+            sys.stdout.buffer.flush()
+
+        for chunk in stream_subprocess_output(cmd):
```

Comment:
I guess I meant a little more of a conceptual yield than "make it into a generator", but this works too 😅